src/remover.rs

```diff
@@ -0,0 +1,157 @@
+use anyhow::{bail, Result};
+use rnix::{SyntaxNode, TextRange};
+
+pub fn remove_dep(
+    contents: &mut String,
+    deps_list: SyntaxNode,
+    remove_dep_opt: Option<String>,
+) -> Result<String> {
+    let remove_dep = match remove_dep_opt {
+        Some(remove_dep) => remove_dep,
+        None => bail!("error: no dependency to remove"),
+    };
```

Comment:
```suggestion
    let remove_dep = match remove_dep_opt.context("finding dependency to remove")?;
```

You can remove some boilerplate on the `Option` matches by using `.context(msg)` to turn the `Option` into a `Result` which you can then use `?` on.

(same for other cases of this)